packages/eslint-plugin/tests/rules/prefer-enum-initializers.test.ts

```diff
@@ -0,0 +1,269 @@
+import rule from '../../src/rules/prefer-enum-initializers';
+import { RuleTester } from '../RuleTester';
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('prefer-enum-initializers', rule, {
+  valid: [
+    `
+enum Direction {}
+    `,
+    `
+enum Direction {
+  Up = 1,
+}
+    `,
+    `
+enum Direction {
+  Up = 1,
+  Down = 2,
+}
+    `,
+    `
+enum Direction {
+  Up = 'Up',
+  Down = 'Down',
+}
+    `,
+  ],
+  invalid: [
+    {
+      code: `
+enum Direction {
+  Up,
+}
+      `,
+      errors: [
+        {
+          messageId: 'defineInitializer',
+          data: { name: 'Up' },
+          line: 3,
+          suggestions: [
+            {
+              messageId: 'defineInitializerSuggestion',
+              /* eslint-disable @typescript-eslint/internal/plugin-test-formatting */
+              output: `
+enum Direction {
+  Up = 0,
+}
+      `,
+              /* eslint-enable @typescript-eslint/internal/plugin-test-formatting */
+            },
+            {
+              messageId: 'defineInitializerSuggestion',
+              /* eslint-disable @typescript-eslint/internal/plugin-test-formatting */
+              output: `
+enum Direction {
+  Up = 1,
+}
+      `,
+              /* eslint-enable @typescript-eslint/internal/plugin-test-formatting */
+            },
+            {
+              messageId: 'defineInitializerSuggestion',
+              /* eslint-disable @typescript-eslint/internal/plugin-test-formatting */
+              output: `
+enum Direction {
+  Up = 'Up',
+}
+      `,
+              /* eslint-enable @typescript-eslint/internal/plugin-test-formatting */
+            },
```

Comment:
This rule is a bit clunky for suggestions, sorry. I need to find a better solution for it.
For now you can just do this instead.

```ts
      code: `
enum Direction {
  Up,
}
      `.trimRight(),
      errors: [
        {
          messageId: 'defineInitializer',
          data: { name: 'Up' },
          line: 3,
          suggestions: [
            {
              messageId: 'defineInitializerSuggestion',
              output: `
enum Direction {
  Up = 0,
}
              `.trimRight(),
            },
            {
              messageId: 'defineInitializerSuggestion',
              output: `
enum Direction {
  Up = 1,
}
              `.trimRight(),
            },
            {
              messageId: 'defineInitializerSuggestion',
              output: `
enum Direction {
  Up = 'Up',
}
              `.trimRight(),
            },
```